test/client_test.py

```diff
@@ -299,6 +349,7 @@ def test_upload_shallow(srv, nbd_server, tmpdir, base_fmt):
         src_base,
         url,
         srv.config.tls.ca_file,
+        disk_is_zero=disk_is_zero,
```

Comment:
But this is uploading a disk that is not zero, and this parameter can be True.
Wouldn't that be wrong usage according to the docstring?
```
Should used only when uploading to new disk on file based storage or new
sparse disk on block based storage.
```

Reply:
How is the parameter tested here? Through the qemu-img compare? As in: the resulting disk will be the same as the source image regardless of the disk being zero or not.

Reply:
We test that the content is correct with qemu-img compare.

For shallow upload the value does not matter since we set backing_chain to False. The test verify that setting it to True does not break the upload.